### ISSUE TITLE
Each child in an array should have a unique "key" prop

### ISSUE THREAD

**Issue by kevinresol** [MVCoconut/coconut.react-core#3]
```jsx
<Parent>
  <Child1/>
  <Child2/>
</Parent>
```

Currently the above hxx generates something like this:

```js
var props = {
  children: [
    React.createElement(Child1),
    React.createElement(Child2),
  ]
}
React.createElement(Parent, props);
```

Note that children is an array, that's why React complains the lack of `key`.

Instead, when the children elements are known at compile-time, it should generate this:

```js
var props = {};
React.createElement(Parent, props, React.createElement(Child1), React.createElement(Child2));
```

So that the children is passed directly to the createElement function as rest arguments.

**Comment by kevinresol** [comment]
The fix is to spread the children into the `createElement` function:

ES6:
`React.createElement(Parent, props, ...children);`

or plain JS:
`React.createElement.apply(React, [Parent, props].concat(children));`

**Comment by kevinresol** [comment]
poc: https://github.com/MVCoconut/coconut.react/commit/5f76eefdfa1bfc054a511e13aedb6c86943cd36d

**Comment by kevinresol** [comment]
I pushed a poc, but it works on react native components but not on coconut views

**Comment by kevinresol** [comment]
I hope this [Babel example](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&code_lz=G4QwTgBAxg9gtgBwJYBsCmAXJc0QLwAUAUBBADwAmSwAfCaeQM4IgB2NAKmgB4YCMZAPTM2dBkxbsuvAExCR7ekKq0iASgDcRIqEhIMaOI3wQA2gF0tu6DACurDCb4AGLQDMYYAtaQnXEXzIbewwNAIBqcLUAgyMAOgRbRgALAjIFGgBvJABfeUkaTW1rMBDsXEIlFTEGbNjGHIglQWr1DSA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=) would explain how it works

**Comment by back2dos** [comment]
Is this still a thing? Or do I need to fix #7 first to get the warning?

**Comment by kevinresol** [comment]
This should be done now https://github.com/MVCoconut/coconut.react-core/blob/3026f61/src/coconut/react/macros/Setup.hx#L100
